hydra-main/src/main/java/com/addthis/hydra/job/spawn/search/JobSearcher.java

```diff
@@ -167,19 +167,37 @@ private SearchResult getMacroSearchResult(String macroName, Set<TextLocation> ma
 
             // See if any of the (recursive) dependencies of this macro had search results. If they do, we add a new
             // LineMatch to this macro's search results, which indicates where the macro w/ a search result was included
-            macroSearchResults.addAll(getDependencySearchMatches(macroIncludeLocations, results));
+            macroSearchResults.addAll(getDependencySearchMatches(macroIncludeLocations, dependencyGraph, results));
         }
 
         return results;
     }
 
+
+    /**
+     * Returns a set of TextLocations which represent that an included dependency (or one of the included dependency's
+     * dependencies, etc.) contains a search match
+     *
+     * @param macroIncludeLocations
+     * @param dependencyGraph
+     * @param macroSearchResults
+     * @return
+     */
     private ImmutableSet<TextLocation> getDependencySearchMatches(MacroIncludeLocations macroIncludeLocations,
+                                                                  JobMacroGraph dependencyGraph,
                                                                   Map<String, Set<TextLocation>> macroSearchResults) {
+
         ImmutableSet.Builder<TextLocation> builder = ImmutableSet.builder();
         for (String depMacroName : macroIncludeLocations.dependencies()) {
-            Set<TextLocation> depMacroResults = macroSearchResults.getOrDefault(depMacroName, ImmutableSet.of());
-            if (!depMacroResults.isEmpty()) {
-                builder.addAll(macroIncludeLocations.locationsFor(depMacroName));
+
+            // For each dependency that depMacroName brings in, see if any of THEM have search results.  If they do, we
+            // want to link back to the include to depMacroName in the search result.
+            for (String deeperDepName : dependencyGraph.getDependencies(depMacroName)) {
```

Comment:
enclosing for loop could be removed if jobs, macros, and aliases were all stored in the same dependency graph